### PR TITLE
Use the initialisation procedure used by DeepMind for testing

### PR DESCRIPTION
**Pull request by alito**
In their paper titled "Massively Parallel Methods for Deep Reinforcement Learning", DeepMind described more precisely how the testing scores were obtained in their Nature paper. They issue null actions for a random number of frames at the start of each test episode so as to break the deterministic behaviour of the emulator. 

This patch implements that

### CODE REVIEW

**Comment by spragunr** [comment]
Thanks for working on this.  It's been on my TODO list, but I haven't gotten around to it.  A couple of thoughts: 

Looking over the Deepmind code, I'm pretty sure that they are measuring these null-actions in units of "frames", which means that your code is effectively performing four times as many null-actions as theirs.  (One of the reasons I haven't worked on this is that the Deepmind implementation is pretty hard to follow.  The logic is divided across three libraries: alewarp, xitari and the Q-network.)  They can do this because they don't use the frame-skip flag.  They handle the frame skipping manually, which also makes it possible for them to do the custom max merging of subsequent frames.  I'm really not sure if either of those issues make much difference, but for the sake of consistency, it might make sense to make both changes at once.  

I think that the random null-ops should happen during both training and testing.

The options now are: 

1. Merge a version of the current pull request with the null-ops during both training and testing, and a reduced max null-op number for better consistency. 

2. Hold off on this until someone has time to make both changes (random null-ops and max-merged frames).

I'm leaning toward 1.  I won't have much time to work on this over the next couple of days, and manually implementing the frame skip may need to be done carefully to avoid slowing down the code. 

Also... I'm trying to keep the code reasonably PEP8 compliant in terms of line length, formatting etc.  It would save me some effort in merging if you could run pylint before submitting pull requests.

**Comment by alito** [comment]
Ok, I've modified it to run during both testing and training, and shortened it to 8 (although I see this second change as moving it away from DeepMind in intention even if it makes it closer in frame length because it reduces the number of different starting positions).

I've fixed the long lines. I do usually run it against pylint, but I've got the convention rules turned off. I've run it against pep8 instead. It spots other issues too, tell me if you'd rather I fix those too (eg "too many blank lines (2)"  and "at least two spaces before inline comment")

**Comment by spragunr** [comment]
I ended up working from the previous pull request to implement manual frame skipping.  There is a slight performance penalty, but I don't think it's any worse than using the existing --merge flag.  The new behavior is the default.  This will prevent mysterious bad performance on games with flicker like space invaders.  The manual frame skipping also improves random starts as discussed above.
